contxt/models/__init__.py

```diff
@@ -85,140 +73,12 @@ def datetime(dt: _datetime, strict: bool = True) -> str:
-        pass
-
-    @property
-    def creatable_fields(self) -> Dict[str, ApiField]:
-        cls = type(self)
-        if not hasattr(cls, "_creatable_fields"):
-            cls._creatable_fields = {
-                f.attr_key: f for f in cls._api_fields if f.creatable
-            }
-        return cls._creatable_fields
-
-    @property
-    def updatable_fields(self) -> Dict[str, ApiField]:
-        cls = type(self)
-        if not hasattr(cls, "_updatable_fields"):
-            cls._updatable_fields = {
-                f.attr_key: f for f in cls._api_fields if f.updatable
-            }
-        return cls._updatable_fields
-
-    @classmethod
-    def from_api(cls, api_dict: Dict):
-        api_dict = deepcopy(api_dict)
-        # Create clean dictionary to pass to init
-        clean_dict = {}
-        for field in cls._api_fields:
-            if field.api_key not in api_dict and not field.optional:
-                raise KeyError(
-                    f"Required API field '{field.api_key}' is missing from"
-                    f" response: {api_dict}"
-                )
-            clean_dict[field.attr_key] = cls.clean_api_value(
-                field=field, value=api_dict.pop(field.api_key, None)
-            )
-
-        # Warn of any unused keys
-        for k in api_dict.keys():
-            logger.warning(f"{cls.__name__}: Unexpected key from api {k}")
-
-        # Return new instance
-        return cls(**clean_dict)
-
-    @classmethod
-    def clean_api_value(cls, field: ApiField, value: Any):
-        if value is None:
-            # No value
-            return value
-        elif isinstance(value, (list, tuple)):
-            # Value is a list, clean each item
-            return [cls.clean_api_value(field, v) for v in value]
-        # elif issubclass(field.data_type, ApiObject):
-        elif callable(getattr(field.data_type, "from_api", None)):
-            # Type is an ApiObject, apply from_api instead of init
-            return field.data_type.from_api(value)
-        else:
-            # Apply type
-            return field.data_type(value)
-
-    def post(self):
-        """Gets data for a POST request"""
-        # Transform api fields to dict
-        d = Serializer.to_dict(
-            self, key_filter=lambda k: k in set(self.creatable_fields.keys())
-        )
-        # Swap attr_keys for api_keys
-        return {self.creatable_fields[k].api_key: v for k, v in d.items()}
-
-    def put(self):
-        """Gets data for a PUT request"""
-        # Transform api fields to dict
-        d = Serializer.to_dict(
-            self, key_filter=lambda k: k in set(self.updatable_fields.keys())
-        )
-        # Swap attr_keys for api_keys
-        return {self.updatable_fields[k].api_key: v for k, v in d.items()}
+    @staticmethod
+    def pascal_case(text: str) -> str:
+        # TODO
+        if " " not in text:
+            return text
+        return text.title().replace(" ", "")
```

Comment:
MIght be cleaner and easier to read to just always do the text.title().replace(" ", "") since replace just looks for blank space anyway.  Seems like a redundant check